include/wabt/binary-reader.h

```diff
@@ -52,13 +52,20 @@ struct ReadBinaryOptions {
   bool skip_function_bodies = false;
 };
 
-// TODO: Move somewhere else?
+// TODO: Move both TypeMut and GCTypeInformation somewhere else?
 struct TypeMut {
   Type type;
   bool mutable_;
 };
 using TypeMutVector = std::vector<TypeMut>;
 
+// Garbage Collector specific type information
```

Comment:
This is a core part of the patch. It contains the `(sub ...)` part of the type. It is declared as a structure, because it is not mandatory.